GitExtUtils/HexString.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Text;
+
+namespace GitCommands
+{
+    public static class HexString
+    {
+        /// <summary>
+        /// Converts a byte array into its hex representation.
+        /// </summary>
+        public static string FromByteArray(byte[] data, bool lowerCase)
```

Comment:
This has two callsites, and both use `lowerCase: true`. With this I suggest dropping `lowerCase` arg, and simplify the logic.
I quickly searched the code, we do use "X2" in few places, but the code will need to be refactored to use this method, and it is outside the scope of this PR.

Reply:
I added this parameter (and intentionally didn't make it an optional parameter) because I wanted it to be a conscious decision a developer has to make if he wants lower or uppercase hex (Gravatar for example doesn't work with uppercase)

I can change that if you want me to, but to _"simplify the logic"_ in this case means removing a single (IMHO trivial) line of code and sacrificing the fact that a developer wouldn't know what (s)he gets unless (s)he tries it or looks into the source - which wouldn't make it simpler because everyone would have to look at the source to see if it suits their needs.

Reply:
> I added this parameter (and intentionally didn't make it an optional parameter) because I wanted it to be a conscious decision a developer has to make if he wants lower or uppercase hex (Gravatar for example doesn't work with uppercase)

Right now we have only a need for a lower-case string. And we can address the next point:

> the fact that a developer wouldn't know what (s)he gets unless (s)he tries it or looks into the source - which wouldn't make it simpler because everyone would have to look at the source to see if it suits their needs.

...with an xml-doc, which will be translate into an intellisense.

When we have a use case requiring the upper case we'll update the code to handle it. 

If we were building an SDK future-proofing API makes sense, here we (still) in control of our API, and we can change it as we need it.
One thing I learnt building SDKs, exposing too much API in a long run makes it harder to evolve an SDK.

/cc: @ivangrek I'd leave this implementation with the app, and won't make it universally available.

Reply:
✔ done - removed the lowercase parameter